src/database-controller/src/core/util.js

```diff
@@ -0,0 +1,84 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+const logger = require('@dbc/core/logger');
+
+async function timePeriod(ms) {
+  await new Promise((resolve, reject) => {
+    setTimeout(() => resolve(), ms);
+  });
+}
+
+function alwaysRetryDecorator(
+  promiseFn,
+  loggingMessage,
+  initialRetryDelayMs = 500,
+  backoffRatio = 2,
+  maxRetryDelayMs = 120000,
```

Comment:
>120000 [](start = 20, length = 6)

2min too long, max 1min?
Better to add rand, such as nextDelayMs * (1 + rand(-0.8, 1))

Reply:
It is used mainly for error handling. 2 min is OK.

Reply:
add rand

Reply:
Done. And an `randomizeDelay` parameter. When it is true, `nextDelayMs = nextDelayMs * (Math.random() * 0.4 + 0.8)` will be executed.